src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -176,6 +178,15 @@ public ResponseEntity<JobModel> deleteDataset(@PathVariable("id") UUID id) {
     return jobToResponse(jobService.retrieveJob(jobId, userReq));
   }
 
+  @Override
+  public ResponseEntity<DatasetSummaryModel> patchDataset(
+      UUID id, DatasetPatchRequestModel patchRequest) {
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    iamService.verifyAuthorization(
+        userReq, IamResourceType.DATASET, id.toString(), IamAction.MANAGE_SCHEMA);
```

Comment:
Both dataset [stewards](https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf#L1060) and [custodians](https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf#L1063) have the `mangage_schema` action in SAM. I think it makes sense that a data custodian would want the ability to PATCH dataset information (especially when we include more fields) but updating the PHS Id indirectly affects who has access to the data.

Right now only dataset stewards and admins are allowed to update who has access to a dataset in SAM (they have the `share_policy::<role>` actions). This change would allow custodians to indirectly change who has read access, but I think that's okay because they still can't add other custodians or stewards. What do you think?

Reply:
I'm glad that you picked up on this nuance.  Thanks as well for the links to the sam configurations!

What could work here is conditionally checking for some `share_policy` authorization in addition to the `manage_schema` check, in cases where we are modifying passport identifiers.  I may ping you about this to chat further though: I see we register this authorization as `IamAction.SHARE_POLICY` without the role qualifications, and the top-level `IamAction` comment reads as follows:

> // NOTE: these action enums must have exactly the same text as the Sam action name.

Reply:
Thanks for making these changes! We actually want to check if the user has the `share_policy::reader` action because that means they are allowed to add readers to the dataset

Reply:
This is definitely an interesting case - Thanks for thinking about this! I don't think I would have caught it.

Two things to think about:
With both MANAGE_SCHEMA and SHARE_POLICY_STEWARD, if I'm understanding correctly, we're saying that a user can alter the PHS ID in two cases:
1. Have steward role
2. Have Custodian AND Admin Role (custodian provides manage schema permission and admin provides the share policy permission)

I think #2 is problematic. As far as I know, we don't usually allow combining roles to get to the right permissions. It would be better to add a new SAM action and give it direct access to the particular roles we've determined. 

I also think this will be an issue when we add additional fields to the update. For example, with the "additional properties" field we're about to add to the dataset patch, the user doesn't need the share_policy permission. Therefore, Custodians will be blocked from making the update, when they really should have permission.

Reply:
@samanehsan

RE: `share_policy::reader`:

I thought that role sounded right at first, but this is a share policy associated with snapshots and not datasets.  I confirmed this manually by checking for `share_policy::reader` on my test dataset in [sam dev API](https://sam.dsde-dev.broadinstitute.org/#/Resources/getPolicyV2), it returned false.

Here are the share policies granted to a [dataset steward](https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf#L1060):
```
"share_policy::steward", "share_policy::custodian", "share_policy::ingester", "share_policy::snapshot_creator"
```

Reply:
@snf2ye 

RE: stacking roles to complete needed permissions:

This counterexample helps explain what felt like a funny smell :) initially I didn't realize what control we had over SAM actions.  I'd like to talk to you about what's needed to create a new one, etc.

RE: Adding additional fields to update:

This probably isn't relevant now, but I don't think this would pose a problem, as the share_policy permission is conditionally checked.  We only check it when modifying a passport identifier.  For "additional properties" patching we would only check for manage_schema permission.

Reply:
Oh! You're completely right @okotsopoulos! I mixed up datasets and snapshot policies 🤦‍♀️ That's probably why there was only a `SHARE_POLICY` defined then since the steward is currently the only role that can share datasets. In that case, we should check for all of the `share_policy` actions for datasets. I think we can exclude `ingester`, since it is a deprecated policy (we'll need to add that to our SAM spring-cleaning list).

Reply:
Hmm I see Shelby's point now...updating the SAM config with a new action for patching the PHS ID and giving it to the dataset steward role would fix this.

Reply:
>  don't think this would pose a problem, as the share_policy permission is conditionally checked.
Ah, ok! Makes sense!